sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/models/AttestationOptions.java

```diff
@@ -0,0 +1,157 @@
+ * come from inside the enclave (if, for instance the RunTime Data is an asymmetric cryptographic
+ * key, it can be used to encrypt data that can only be decrypted by code inside the TEE).
+ *
+ * For both InitTime and RunTime data, the data can be expressed in the token as either BINARY or
+ * JSON data. This can simplify the evaluation process in the relying party.
+ *
+ *
+ * In addition to the evidence, InitTime and RunTime data, an Attest request can also take a "draft"
+ * attestation policy. This can be used to determine how an attestation poilcy effects the claims
+ * generated by the attestation service.
+ *
+ * Note that when a client specifies a draft attestation policy the returned attestation token will not
+ * be signed by the attestation service. That is to ensure that the token cannot be used by a relying
+ * party.
+ *
+ * The reason that both binary and JSON are represented as binary is that JSON encodings are ambiguous
+ * – there are many possible semantically identical JSON encodings of the same data. Since we’re going
+ * to take the SHA256 hash of the data, we accept a sequence of octes (byte[]) instead of a String
+ * (depending on your encoding, the same string can be encoded as different sequences of octets).
+ *
+ */
+public interface AttestationOptions {
+    /**
+     * Creates a new AttestOpenEnclaveRequest object with the OpenEnclave report from the enclave to be attested.
+     *
+     * The evidence will typically take the form of either an OpenEnclave report or an Intel SGX quote.
+     *
+     * @param evidence to be used in the attest request.
+     * @return the AttestOpenEnclaveRequest object itself.
+     */
+    static AttestationOptions fromEvidence(byte[] evidence) {
+        return new AttestationOptionsImpl().setEvidence(evidence);
+    }
+
+    /**
+     * Set the RunTime Data property: Runtime data provided by the enclave at the time the evidence
+     * was generated. The attestation service will verify that the first 32 bytes of the report_data
+     * field of the quote contains the SHA256 hash of the decoded "data" field of the runtime data, that
+     * allows the enclave to attest to the correctness of the RunTime data.
+     *
+     * @param runTimeData the runtimeData value to set.
+     * @return the AttestOpenEnclaveRequest object itself.
+     */
+    AttestationOptions setRunTimeData(byte[] runTimeData);
+
+    /**
+     * Retrieves the RunTimeData property to be sent to the service.
+     * @return The RunTimeData value set by {@link AttestationOptions#setRunTimeData}
+     */
+    byte[] getRunTimeData();
+
+    /**
+     * When the attestation service receives the RunTime Data, it should interpret the data as
+     * binary, and return the data in the {@link AttestationResult#getEnclaveHeldData()} property
+     * @return the AttestationOptions object.
+     */
+    AttestationOptions interpretRunTimeDataAsBinary();
```

Comment:
If you are returning AttestationOptions object, then this method is still essentially a fluent setter, so we might as well name it that.

Reply:
But it's *not* a setter - we had that discussion in email. It's an assertation about how the value should be interpreted by the service. There is no "getter" that corresponds to this (it's a two state toggle switch).